pkg/datarepair/checker/checker.go

```diff
@@ -217,3 +169,115 @@ func contains(a []string, x string) bool {
 	}
 	return false
 }
+
+func (checker *Checker) checkSegmentStatus(ctx context.Context, pointer *pb.Pointer, path string, monStats *durabilityStats) (err error) {
```

Comment:
the naming of this function may have drifted over time.  it not only checks the segment status but also records the status in the database.  Consider renaming to something more generic, maybe 'updateSegmentStatus'?

Reply:
updated the function name as suggested.